extensions/canvasMedical/actions/getQuestionnaireResponse.ts

```diff
@@ -0,0 +1,110 @@
+import { z, ZodError } from 'zod'
+import {
+  FieldType,
+  type Action,
+  type DataPointDefinition,
+  type Fields,
+} from '@awell-health/extensions-core'
+import { Category } from '@awell-health/extensions-core'
+import { type settings } from '../settings'
+import { makeAPIClient } from '../client'
+import type schemas from '../schemas'
+import { fromZodError } from 'zod-validation-error'
+import { AxiosError } from 'axios'
+
+const fields = {
+  questionnaireResponseId: {
+    id: 'questionnaireResponseId',
+    label: 'Questionnaire Response Id',
+    description: 'The FHIR resource id for the QuestionnaireResponse resource',
+    type: FieldType.STRING,
+    required: true,
+  },
+} satisfies Fields<typeof schemas>
+
+const dataPoints = {
```

Comment:
**Note for the future, this action isn't being exported so this isn't feedback on the action itself**

This is an example of the full response object when reading a questionnaire response: 
```
{
  "resourceType": "QuestionnaireResponse",
  "id": "c1ca7b42-b36d-45ef-8e0d-c39f5d27d179",
  "meta": {
    "versionId": "1",
    "lastUpdated": "2021-09-20T13:45:25.229+00:00"
  },
  "questionnaire": "Questionnaire/ac1da1a4-ccc4-492e-a9e0-7f70a58c2129",
  "status": "completed",
  "subject": {
    "reference": "Patient/69dd2c8eb9244ff39c178b3c1f847045"
  },
  "authored": "2021-09-16T19:09:29.530419+00:00",
  "author": {
    "reference": "Practitioner/4150cd20de8a470aa570a852859ac87e"
  },
  "item": [
    {
      "linkId": "2021b11e-b90f-465c-bd16-70b98779272a",
      "text": "Tobacco type",
      "answer": [
        {
          "valueCoding": {
            "system": "http://snomed.info/sct",
            "code": "722496004",
            "display": "Cigarettes"
          }
        },
        {
          "valueCoding": {
            "system": "http://snomed.info/sct",
            "code": "722498003",
            "display": "eCigarette"
          }
        }
      ]
    },
    {
      "linkId": "6359d061-0820-4c36-950c-7a0f996eda98",
      "text": "Tobacco status",
      "answer": [
        {
          "valueCoding": {
            "system": "http://snomed.info/sct",
            "code": "428061000124105",
            "display": "Current Light tobacco user"
          }
        }
      ]
    },
    {
      "linkId": "94651467-f7c5-45d5-b24e-3b2bc42c79c0",
      "text": "Tobacco comment",
      "answer": [
        {
          "valueString": "Would like information about quitting"
        }
      ]
    }
  ]
}
```

https://docs.canvasmedical.com/reference/questionnaireresponse-read

As you can see there are a lot of fields, but most importantly, there is the `item` field which is an array. Without a mechanism to know what is in this array, we won't be able to specify data points to create apart from the static ones in this object (e.g. `status`). If we are not able to extract the answers then the value of this action is very limited